frontend/src/common/queries/wheresMyGene.ts

```diff
@@ -628,13 +862,17 @@ function clobberQueryIfSubsetOfPrev(
   filtersToCheck: (keyof Filter)[]
 ): Query | null {
   if (prevQuery == query) return prevQuery;
+
   if (
     !prevQuery ||
     prevQuery?.include_filter_dims !== query?.include_filter_dims
   ) {
     prevQuery = query;
     return query;
   }
+
+  if (prevQuery.compare !== query.compare) return query;
```

Comment:
this is added to ensure `clobberQueryIfSubsetOfPrev` doesn't use the cached query incorrectly